libvast/vast/table_slice_row.hpp

```diff
@@ -0,0 +1,69 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/fwd.hpp"
+#include "vast/view.hpp"
+
+#include <caf/meta/type_name.hpp>
+
+#include <cstdint>
+#include <string_view>
+
+namespace vast {
+
+/// Convenience helper for traversing a row of a table slice.
+/// @relates table_slice
+class table_slice_row {
```

Comment:
What is the purpose of this class? I think it would be preferable to use `record_view` instead.

Reply:
We have a story for a future expansion that makes me want to keep this as is.

Reply:
Can you be more specific? What functionality should be based on this?

Reply:
Row-major and column-major iteration can be built on top of these.

Reply:
this type would only be useful for row-major iteration, but that iterator should point to `record_view`s.

Reply:
That would require decoding the whole row beforehand, which we don't necessarily want to do. The data access should be lazy through the encoding, regardless of whether it is MessagePack or Arrow.

Reply:
Clearly, we don't have the use case fleshed out yet. Until then, I propose we put these utilities in the backlog and add them only when needed. Otherwise we carry more complexity with us that we need.

Reply:
We agreed to keep the type around in case we really do need it later. It is much more discoverable inside the tree than somewhere in an old branch.